### PR TITLE
Fix output_prefix in do() method for ChatGPT Agent

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Tracking issue
https://github.com/flyteorg/flyte/issues/3936

## Why are the changes needed?
ChatGPT Agent will fail if we don't merge this PR.
(I've added a unit test to prevent the same error in the future.

## What changes were proposed in this pull request?
1. Add the ~~`output_prefix`~~ `**kwargs` field for ChatGPT `do` function.
2. Change the order of `output_prefix` in `SyncAgentBase` for better readability.
3. Add a unit test to mock chatgpt task's life cycle.

## How was this patch tested?
unit test, local execution and single binary.

### Setup process
```python
from flytekit import task, workflow, Secret
from flytekitplugins.openai import ChatGPTTask

chatgpt_new_job = ChatGPTTask(
            name="chatgpt",
            openai_organization="org-NayNG68kGnVXMJ8Ak4PMgQv7",
            # openai_key=Secret(group="openai-remote", key="hanru"),
            # openai_organization="xxx",
            chatgpt_config={
                    "model": "gpt-3.5-turbo",
                    "temperature": 0.7,
            },
        )
if __name__ == "__main__":
    print(chatgpt_new_job(message="hi"))
```

### Screenshots
remote execution (single binary)
image

local execution
image

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: flytekit/extend/backend/base_agent.py [hunk 1]
```diff
@@ -259,8 +261,8 @@ async def _do(
         self: PythonTask,
         agent: SyncAgentBase,
         template: TaskTemplate,
-        output_prefix: str,
         inputs: Dict[str, Any] = None,
+        output_prefix: Optional[str] = None,
```
Comment (pingsutw) [inline flytekit/extend/backend/base_agent.py:265]:
```suggestion
        **kwargs,
```

Comment (Future-Outlier) [reply flytekit/extend/backend/base_agent.py:267]: Nice advice, thank you!!

File: flytekit/extend/backend/base_agent.py [hunk 2]
```diff
@@ -119,7 +119,9 @@ class SyncAgentBase(AgentBase):
     name = "Base Sync Agent"
 
     @abstractmethod
-    def do(self, task_template: TaskTemplate, inputs: Optional[LiteralMap], output_prefix: str, **kwargs) -> Resource:
+    def do(
+        self, task_template: TaskTemplate, inputs: Optional[LiteralMap], output_prefix: Optional[str], **kwargs
+    ) -> Resource:
```
Comment (pingsutw) [inline flytekit/extend/backend/base_agent.py:124]:
```suggestion
    def do(
        self, task_template: TaskTemplate, inputs: Optional[LiteralMap], output_prefix: str, **kwargs
    ) -> Resource:
```

File: flytekit/extend/backend/base_agent.py [hunk 3]
```diff
@@ -259,8 +263,8 @@ async def _do(
         self: PythonTask,
         agent: SyncAgentBase,
         template: TaskTemplate,
-        output_prefix: str,
         inputs: Dict[str, Any] = None,
+        output_prefix: Optional[str] = None,
```
Comment (pingsutw) [inline flytekit/extend/backend/base_agent.py:267]: output_prefix is not optional type

File: plugins/flytekit-huggingface/setup.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 plugin_requires = [
     "flytekit>=1.3.0b2,<2.0.0",
-    "datasets>=2.4.0",
+    "datasets>=2.4.0,<2.19.2",
```
Comment (pingsutw) [inline plugins/flytekit-huggingface/setup.py:9]: why do we change it here

**Review by pingsutw — APPROVED** [review]

**Pull request merged**